### ISSUE TITLE
Create a script that finds inconsistencies between the doc and the current methods/classes

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/ivadomed#310]
## Context

Currently, the documentation is semi-manually maintained: we need to manually write the class/method name to appear in the documentation.

## Problem

- someone adds a new function and forgets to expose it in the docs
- someone removes a function and forgets to remove it in the docs

## Potential solution

a script that would crawls across all of our files, parse classes and methods, and check if there is a one-to-one matching with the entries in the doc.

**Comment by Drulex** [comment]
Since converting the documentation to `rst`, we are now using the `autodoc` extension (https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html) which automatically generates API documentation for classes/methods. At the moment we only need to manually add new modules to the `api.rst` file since we are using `automodule`. Once a module is added, the contents (classes/methods) are automatically updated on code change.

This solves the 2 problems mentioned above, but is still semi-manual (although much less manual than before). Do you think this is enough?

**Comment by jcohenadad** [comment]
> This solves the 2 problems mentioned above, but is still semi-manual (although much less manual than before). Do you think this is enough?

I agree this is much better than before. Let's keep this issue as "low priority", and see how contributors deal with updating what should be updated in the documentation. If there is repetitive mismatch between the doc and new features added, maybe we could revisit the "checking" script.
